premium/usage.go

```diff
@@ -0,0 +1,240 @@
+package premium
+
+import (
+	"context"
+	"fmt"
+	cqapi "github.com/cloudquery/cloudquery-api-go"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+	"net/http"
+	"sync/atomic"
+	"time"
+)
+
+const (
+	defaultBatchLimit  = 1000
```

Comment:
This might be a bit tricky to implement, but I think we may also want to have a minimum time between flushes, probably around 10 or 20 seconds. This way we don't risk sending a thousand requests in all quick succession if we're able to sync 1M rows in a short period.

Reply:
Added a comparison to the last update time here 59e4ab28f23cc93ab2acc49bdb2024d3c969dab2, which will prevent updates happening quicker than a minimumUpdateDuration (default 10s).